src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System.Collections;
+using Xunit;
+
+namespace System.DirectoryServices.Tests
+{
+    public class DirectoryServicesTests
+    {
+        internal static bool IsLdapConfigurationExist => LdapConfiguration.Configuration != null;
+        
+        [ConditionalFact(nameof(IsLdapConfigurationExist))]
+        public void TestOU() // adding and removing organization unit
+        {
+			using (DirectoryEntry de = new DirectoryEntry(LdapConfiguration.Configuration.LdapPath,
+														    LdapConfiguration.Configuration.UserName,
+														    LdapConfiguration.Configuration.Password,
+														    LdapConfiguration.Configuration.AuthenticationTypes))
+            {
+                string ouName = "NetCoreDevs";
+                CreateOU(de, ouName, ".Net Core Developers Unit");
```

Comment:
You should try and remove any pre-exisiting OUs with the target name first to take care of tests which leave the environment dirty

Reply:
Thanks. I'll fix that